src/Integration.Vsix/CFamily/CLangAnalyzer.cs

```diff
@@ -99,33 +105,38 @@ private async Task TriggerAnalysisAsync(Request request, IIssueConsumer consumer
             await TaskScheduler.Default;
 
             logger.WriteLine($"Analyzing {request.File}");
-            var messages = new List<Message>();
-            Action<Message> handleMessage = message => messages.Add(message); // todo: change to consumer.Accept
+            int issueCount = 0;
+            Action<Message> handleMessage = message => HandleMessage(message, request, consumer, ref issueCount);
 
             // We're tying up a background thread waiting for out-of-process analysis. We could
             // change the process runner so it works asynchronously. Alternatively, we could change the
             // RequestAnalysis method to be asynchronous, rather than fire-and-forget.
-            CFamilyHelper.CallClangAnalyzer(handleMessage, request, new ProcessRunner(settings, logger), analysisStatusNotifier, logger, cancellationToken);
-
-            Debug.Assert(messages.All(m => m.Filename == request.File), "Issue for unexpected file returned");
-
-            var issues = messages
-                .Where(m => IsIssueForActiveRule(m, request.RulesConfiguration))
-                .Select(m => CFamilyHelper.ToSonarLintIssue(m, request.CFamilyLanguage, request.RulesConfiguration))
-                .ToList();
+            CallSubProcess(handleMessage, request, analysisStatusNotifier, settings, logger, cancellationToken);
 
             telemetryManager.LanguageAnalyzed(request.CFamilyLanguage); // different keys for C and C++
 
-            logger.WriteLine($"Found {issues.Count} issue(s)");
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                logger.WriteLine($"Found {issueCount} issue(s) for {request.File}");
+            }
+        }
+
+        private void HandleMessage(Message message, Request request, IIssueConsumer consumer, ref int issueCount)
+        {
+            Debug.Assert(message.Filename == request.File, $"Issue for unexpected file returned: {message.Filename}");
+            if (!IsIssueForActiveRule(message, request.RulesConfiguration))
+            {
+                return;
```

Comment:
According to SonarCloud there are 3uncovered lines of code out of 7, giving 57% coverage. This is one of the uncovered lines, which I can add a test for quite easily.

The other two are for the new virtual method declaration and the one-line passthough call to CFamilyHelper. Testing that would require an integration tests, which I'm not proposing to do at this point.

Reply:
I agree. It's worth testing the non-active rule logic.